tony-core/src/main/java/com/linkedin/tony/TonyClient.java

```diff
@@ -665,7 +672,12 @@ private void printTaskUrls() throws IOException, YarnException {
     if (amRpcServerInitialized && !hasPrintedTaskUrls) {
       Set<TaskUrl> taskUrls = amRpcClient.getTaskUrls();
       if (!taskUrls.isEmpty()) {
-        new TreeSet<TaskUrl>(taskUrls).forEach(task -> Utils.printTaskUrl(task, LOG));
+        new TreeSet<TaskUrl>(taskUrls).forEach(task -> {
+            if (task.getName().equals(Constants.NOTEBOOK_JOB_NAME)) {
+                notebookUrl = task.getUrl();
```

Comment:
This `printTaskUrls()` method shouldn't have side effects like setting a `notebookUrl` field. Please do this in a separate method.

Reply:
yah.. this was a hack, will clean this up.